src/uct/ib/dc/dc_mlx5.h

```diff
@@ -35,6 +35,8 @@ struct ibv_ravh {
 
 #define UCT_DC_MLX5_KEEPALIVE_NUM_DCIS  1
 #define UCT_DC_MLX5_IFACE_MAX_DCI_POOLS 16
+#define UCT_DC_MLX5_IFACE_MAX_DCI_NUM   127
+#define UCT_DC_MLX5_MAX_DCI_NUM         255
```

Comment:
We can have up to 16 dci pools

Reply:
Yeah, we defined the maximum is 16.
But we use `ep->flags & 7` to get pool index, which might result in some problems.
see [`src/uct/ib/dc/dc_mlx5_ep.h](https://github.com/openucx/ucx/blob/master/src/uct/ib/dc/dc_mlx5_ep.h#L247)`:
```
enum uct_dc_mlx5_ep_flags {
    /* DCI pool EP assigned to according to it's lag port */
    UCT_DC_MLX5_EP_FLAG_POOL_INDEX_MASK     = UCS_MASK(3),
    /* … */
}
 

static UCS_F_ALWAYS_INLINE uint8_t
uct_dc_mlx5_ep_pool_index(uct_dc_mlx5_ep_t *ep)
{
    return ep->flags & UCT_DC_MLX5_EP_FLAG_POOL_INDEX_MASK;
}
```

Reply:
this is another issue, but it's better to calculate `UCT_DC_MLX5_MAX_DCI_NUM` based on `UCT_DC_MLX5_IFACE_MAX_DCI_NUM` and number of pools

Reply:
unhandled

Reply:
done.
But the total DCI_NUM still cannot exceed 255 as it's stored in `uint8_t`.